### PR TITLE
Fix wrong position of TabBarIndicator when it's label size and has label padding

### PR DESCRIPTION
**Pull request by zmtzawqlp**
Reopen from #109605

wrong position of TabBarIndicator when set indicatorSize to TabBarIndicatorSize.label and it has labelPadding.

simulator_screenshot_6A60CB7F-D587-4A63-B059-EF4E5CDBA542

Related issue:

[#109604](https://github.com/flutter/flutter/issues/109604)
[#63700](https://github.com/flutter/flutter/issues/63700)

*If you had to change anything in the [flutter/tests] repo, include a link to the migration guide as per the [breaking change policy].*

Add test to the case TabBarIndicatorSize.label and has labelPadding

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

**Comment by zmtzawqlp** [comment]
@goderbauer  please take a look if you have a time

File: packages/flutter/lib/src/material/tabs.dart [hunk 1]
```diff
@@ -331,6 +331,7 @@ class _IndicatorPainter extends CustomPainter {
     required _IndicatorPainter? old,
     required this.indicatorPadding,
     this.dividerColor,
+    required this.labelPaddings,
```
Comment (AlexV525) [inline packages/flutter/lib/src/material/tabs.dart:334]: Can you put this before the `dividerColor` to place all required parameters first?

**Review by AlexV525 — APPROVED** [review]

File: packages/flutter/lib/src/material/tabs.dart (hunk 1 above)
Comment (zmtzawqlp) [reply packages/flutter/lib/src/material/tabs.dart:333]: done

**Comment by goderbauer** [comment]
Looks like there's still a merge conflict. Can you fix that one?

File: packages/flutter/lib/src/material/tabs.dart [hunk 2]
```diff
@@ -954,6 +959,7 @@ class _TabBarState extends State<TabBar> {
     // If indicatorSize is TabIndicatorSize.label, _tabKeys[i] is used to find
     // the width of tab widget i. See _IndicatorPainter.indicatorRect().
     _tabKeys = widget.tabs.map((Widget tab) => GlobalKey()).toList();
+    _labelPaddings = widget.tabs.map<EdgeInsetsGeometry>((Widget tab) => EdgeInsets.zero).toList();
```
Comment (goderbauer) [inline packages/flutter/lib/src/material/tabs.dart:962]: I believe, this and the previous line could be simplified if List.filled is used: https://master-api.flutter.dev/flutter/dart-core/List/List.filled.html

File: packages/flutter/lib/src/material/tabs.dart [hunk 3]
```diff
@@ -1092,8 +1099,10 @@ class _TabBarState extends State<TabBar> {
     if (widget.tabs.length > _tabKeys.length) {
       final int delta = widget.tabs.length - _tabKeys.length;
       _tabKeys.addAll(List<GlobalKey>.generate(delta, (int n) => GlobalKey()));
+      _labelPaddings.addAll(List<EdgeInsetsGeometry>.generate(delta, (int n) => EdgeInsets.zero));
```
Comment (goderbauer) [inline packages/flutter/lib/src/material/tabs.dart:1102]: could also use List.filled

**Review by chunhtai — APPROVED** [review]
LGTM, once others' comments are addressed

File: packages/flutter/lib/src/material/tabs.dart (hunk 2 above)
Comment (zmtzawqlp) [reply packages/flutter/lib/src/material/tabs.dart:962]: should i also change the code about _tabKeys ?

Comment (zmtzawqlp) [reply packages/flutter/lib/src/material/tabs.dart]: we can't do the same thing with _tabKeys,  List.filled method fill with same GlobalKey, but widget require unique key.  i have done changes with _labelPaddings.

**Review by chunhtai — COMMENTED** [review]
Can you also update the tabKeys

**Comment by AlexV525** [comment]
> Can you also update the tabKeys

See https://github.com/flutter/flutter/pull/116250#discussion_r1036640589 ?

**Comment by chunhtai** [comment]
> See [#116250 (comment)](https://github.com/flutter/flutter/pull/116250#discussion_r1036640589) ?

Thanks for reminding. look good then

**Comment by chunhtai** [comment]
It looks like the google test is acting weird, can you rebase off the latest master branch to see if that fixes it?

**Comment by zmtzawqlp** [comment]
open new pr https://github.com/flutter/flutter/pull/116398

**Pull request closed without merging**